### PR TITLE
Fix/37095: Tag list show required

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

- Disabled tags are showing as required for "Multi level tags"
### Fixed Issues

$ https://github.com/Expensify/App/issues/37095
PROPOSAL: https://github.com/Expensify/App/issues/37095#issuecomment-1959995099

### Tests

1. In OD as admin enable "People must tag expenses" for some workspace
2. Enable "Use multiple levels of tags"
3. Create/import some multi-level tags and mark at least one as not ```required``` (See the below image)
image
4. Invite user B to the policy as employee
5. As employee go to the workspace chat invited to
6. Request money and enter the amount and click next
7. Observe that only required tag in OD appears as required in ND ( There is a "required" text next to the tag list name)
- [x] Verify that no errors appear in the JS console
### QA Steps

1. In OD as admin enable "People must tag expenses" for some workspace
2. Enable "Use multiple levels of tags"
3. Create/import some multi-level tags and mark at least one as not ```required``` (See the below image)
image
4. Invite user B to the policy as employee
5. As employee go to the workspace chat invited to
6. Request money and enter the amount and click next
7. Observe that only required tag in OD appears as required in ND ( There is a "required" text next to the tag list name)
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[Screencast from 04-03-2024 01:10:37.webm](https://github.com/Expensify/App/assets/129500732/f7cbb9b0-45f7-4a5c-9317-120dcadddc72)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[Screencast from 28-02-2024 09:13:37.webm](https://github.com/Expensify/App/assets/129500732/1b74b729-6fa9-4bcc-9ad5-ba7a42c03043)

</details>

<details>
<summary>iOS: Native</summary>

[Screencast from 04-03-2024 01:04:14.webm](https://github.com/Expensify/App/assets/129500732/2d89779a-4c30-4a0c-8955-a4415c8e3eae)

</details>

<details>
<summary>iOS: mWeb Safari</summary>

[Screencast from 28-02-2024 09:13:04.webm](https://github.com/Expensify/App/assets/129500732/9ada592d-b225-44c2-bf71-77a73fada156)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

[Screencast from 28-02-2024 09:12:27.webm](https://github.com/Expensify/App/assets/129500732/a0400f66-3afb-436a-9418-83c2f2d7fb59)

</details>

<details>
<summary>MacOS: Desktop</summary>

[Screencast from 04-03-2024 00:50:14.webm](https://github.com/Expensify/App/assets/129500732/6cc6fb61-3f75-4c22-9049-d4aa16adb020)

</details>

### CODE REVIEW

**Comment by akinwale** [comment]
> @akinwale Ping to review in case you miss it

Thanks for the bump. I will review today.

**Comment by akinwale** [comment]
@dukenv0307 Please upload test videos or screenshots for Android native, iOS native and Desktop.

**Comment by akinwale** [comment]
@dukenv0307

You also need to clarify your test steps because there are two things that can be disabled when dealing with tag lists: the Required flag for a tag list, or each individual tag when a user clicks on edit tags for a particular tag list.

> 3. Create/import some multi-level tags and mark at least one as disabled

What should be marked as disabled? The Required switch or an individual tag when the user clicks on Edit tags?

> 7. Disabled tags shouldn't appear

What should not appear here? The "Required" text next to the tag name, or each individual disabled tag that corresponds to a tag list?

**Comment by akinwale** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/4160319/2665f874-bb22-48bc-8ef3-9037afdadcaf

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/4160319/0e56e8a4-98c0-4a0b-ad1b-88e933edc3f0

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/4160319/48784674-454a-4de5-87c1-017514201e12

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/4160319/234f2c0a-f0d3-4980-885e-2bc150e3d6ee

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/4160319/c45970cf-d020-4161-a9d1-73e034fd0708

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/4160319/b0a7fe0f-185d-439a-9f36-e250f37f4ff7

</details>

**Review by akinwale — CHANGES_REQUESTED** [review]
Just a couple of things to address: https://github.com/Expensify/App/pull/37388#issuecomment-1974651047 and https://github.com/Expensify/App/pull/37388#issuecomment-1974699405.

**Review by akinwale — APPROVED** [review]
LGTM.

File: src/components/MoneyTemporaryForRefactorRequestConfirmationList.js [hunk 1]
```diff
@@ -763,28 +763,38 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
...
-                    disabled={didConfirm}
-                    interactive={!isReadOnly}
-                    rightLabel={isTagRequired ? translate('common.required') : ''}
-                />
-            ),
-            shouldShow: shouldShowTags,
-            isSupplementary: !isTagRequired,
-        })),
+        ..._.map(policyTagLists, ({name, required}, index) => {
+            const isTagRequired = !isUndefined(required) && required;
```
Comment (MonilBhavsar) [inline src/components/MoneyTemporaryForRefactorRequestConfirmationList.js:767]: Why we removed `canUseViolations`?
And this does not work for single level tags where required is not set for each tag, right?

Comment (MonilBhavsar) [reply src/components/MoneyTemporaryForRefactorRequestConfirmationList.js]: Is it still on HOLD? What PR?

**Review by akinwale — APPROVED** [review]

**Comment by MonilBhavsar** [comment]
Sorry looking
Just want to confirm if we followed this checklist item and retested. @dukenv0307 @akinwale 

> If the main branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the Test steps.

**Comment by akinwale** [comment]
> Sorry looking Just want to confirm if we followed this checklist item and retested. @dukenv0307 @akinwale
> 
> > If the main branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the Test steps.

@MonilBhavsar Yes, I did retest on all platforms.

**Review by MonilBhavsar — APPROVED** [review]
Thanks for confirming!

**Pull request merged**